provider/tags.go

```diff
@@ -57,6 +57,40 @@ func applyTags(
 	}
 	if allTags.IsNull() {
 		delete(ret, "tags")
+		delete(ret, "tagsAll")
+		return ret, nil
+	}
+	ret["tags"] = allTags
+	ret["tagsAll"] = allTags
+
+	return ret, nil
+}
+
+// similar to applyTags, but applied to the `TransformOutputs` method that is run
+// during Read
+func applyTagsOutputs(
+	ctx context.Context, config resource.PropertyMap,
+) (resource.PropertyMap, error) {
+	ret := config.Copy()
+	configTags := resource.NewObjectProperty(resource.PropertyMap{})
+	if t, ok := config["tags"]; ok {
+		configTags = t
+	}
+
+	meta := resource.PropertyMap{}
+	if at, ok := config["tagsAll"]; ok {
+		meta["defaultTags"] = resource.NewObjectProperty(resource.PropertyMap{
+			"tags": at,
+		})
+	}
+
+	allTags, err := mergeTags(ctx, configTags, meta)
```

Comment:
I'd appreciate a little bit of elaboration here on what we're trying to do, and how does it work, I see this:

As written, `allTags = mergeTags(ctx, config.tags, {defaultTags: config.tagsAll})`. 

I *think* if I recall what TransformOutputs is doing this will be assigning config to the tags and tagsAll coming out of the TF provider. This seems to then ignore the current defaultTags on the provider itself as it were. 

Why does this work? I thought perhaps we're not trusting the tags_all output from the provider internals, is it useful here?